cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/ApplicationLifecycleService.java

```diff
@@ -1500,6 +1500,9 @@ public void deleteState(AppStateKey request) throws ApplicationNotFoundException
    */
   public void deleteAllStates(NamespaceId namespaceId, String appName)
       throws ApplicationNotFoundException {
+
+    accessEnforcer.enforce(new ApplicationId(namespaceId.getNamespace(), appName),
```

Comment:
nit: You can use `namespaceId.app(appName)` here instead of creating your own

Reply:
Ack, will change this.